add docs on what privileges are needed for using metricbeat mongodb module (#13932)
elastic/beats: metricbeat/module/mongodb/_meta/docs.asciidoc

The MongoDB metricsets were tested with MongoDB 3.4 and 3.0 and are expected to
work with all versions >= 2.8.

[float]
=== MongoDB Privileges

In order to use the metricsets, the MongoDB user specified in the module configuration needs to have certain https://docs.mongodb.com/manual/core/authorization/#privileges[privileges]

Using different metricsets requires different set of privileges

* `collstats` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#top[`top` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

* `dbstats` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#listDatabases[`listDatabases` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

> ** for each of the databases, also need https://docs.mongodb.com/manual/reference/privilege-action/#dbStats[`dbStats` action] on the https://docs.mongodb.com/manual/reference/resource-document/#database-and-or-collection-resource[`database` resource]

Reviewer: dbStats seems to not working 🤔

Author: ooops my bad, will fix

* `status` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#serverStatus[`serverStatus` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

* `replstatus` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#find[`find`/`listCollections` action] on https://docs.mongodb.com/manual/reference/local-database/[the `local` database resource]

> ** https://docs.mongodb.com/manual/reference/privilege-actions/#collStats[`collStats` action] on https://docs.mongodb.com/manual/reference/local-database/#local.oplog.rs[the `local.oplog.rs` collection resource]

Reviewer: What about the [`replSetGetStatus`](https://docs.mongodb.com/manual/reference/privilege-actions/#replSetGetStatus)?

Author: Ahh missed it. Will add

The MongoDB metricsets were tested with MongoDB 3.4 and 3.0 and are expected to
work with all versions >= 2.8.

[float]
=== MongoDB Privileges

> In order to use the metricsets, the MongoDB user specified in the module configuration needs to have certain https://docs.mongodb.com/manual/core/authorization/#privileges[privileges]

Reviewer 2: ```suggestion
In order to use the metricsets, the MongoDB user specified in the module configuration needs to have certain https://docs.mongodb.com/manual/core/authorization/#privileges[privileges].
```

> Using different metricsets requires different set of privileges

Reviewer 2: ```suggestion
Using different metricsets requires different set of privileges:
```

The MongoDB metricsets were tested with MongoDB 3.4 and 3.0 and are expected to
work with all versions >= 2.8.

[float]
=== MongoDB Privileges

In order to use the metricsets, the MongoDB user specified in the module configuration needs to have certain https://docs.mongodb.com/manual/core/authorization/#privileges[privileges].

We recommend using the https://docs.mongodb.com/manual/reference/built-in-roles/#clusterMonitor[`clusterMonitor` role] to cover all the necessary privileges.

You can use the following command in Mongo shell to create the privileged user (make sure you are using the `admin` db by using `db` command in Mongo shell).

["source","js",subs="attributes"]
----
db.createUser(
    {
        user: "beats",
        pwd: "pass",
        roles: ["clusterMonitor"]
    }
)
----

You can use the following command in Mongo shell to grant the role to an existing user (make sure you are using the `admin` db by using `db` command in Mongo shell).

["source","js",subs="attributes"]
----
db.grantRolesToUser("user", ["clusterMonitor"])

> ----

Reviewer 2: Great :slightly_smiling_face:

* `status` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#serverStatus[`serverStatus` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

* `replstatus` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#find[`find`/`listCollections` action] on https://docs.mongodb.com/manual/reference/local-database/[the `local` database resource]
** https://docs.mongodb.com/manual/reference/privilege-actions/#collStats[`collStats` action] on https://docs.mongodb.com/manual/reference/local-database/#local.oplog.rs[the `local.oplog.rs` collection resource]
** https://docs.mongodb.com/manual/reference/privilege-actions/#replSetGetStatus[`replSetGetStatus` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

* `metrics` requires
** https://docs.mongodb.com/manual/reference/privilege-actions/#serverStatus[`serverStatus` action] on https://docs.mongodb.com/manual/reference/resource-document/#cluster-resource[`cluster` resource]

> In Mongo shell, you can use the following command to create the role that encompass all of the privileges

Reviewer 2: ```suggestion
In Mongo shell, you can use the following command to create the role that encompass all of the privileges.
```

....
db.createRole(
   {
     role: "beatsRole",
     privileges: [
       { resource: { db: "local" }, actions: ["listCollections"] },
       { resource: { db: "local", collection: "oplog.rs" }, actions: ["collStats"] },
       { resource: { cluster: true }, actions: ["top", "listDatabases", "serverStatus", "replSetGetStatus"] },
       // for each of the databases you want to use dbstats metricset with
       { resource: { db: "db_1"}, actions: ["dbStats"]},
       { resource: { db: "db_2"}, actions: ["dbStats"]},

     ],
     roles: []
   }
)
....

> You can then use the following command to create the privileged user

Reviewer 2: ```suggestion
You can then use the following command to create the privileged user.
```

And please review similar lines in other files.

....
db.createUser(
    {
        user: "beats",
        pwd: "pass",
        roles: ["beatsRole"]
    }
)
....

> or grant the role to an existing user

Reviewer 2: ```suggestion
To grant the role to an existing user, you can use the following command.
```
And please review similar lines in other files.
